websocket/websocket-server.go

```diff
@@ -0,0 +1,295 @@
+				if !ok {
+					Response := make(map[string]interface{})
+					Response["request_type"] = "send_message"
+					Response["message"] = "message is required"
+					responseBytes, _ := json.Marshal(Response)
+					WriteOnSocket(responseBytes, messageType, Conn)
+					return
+				}
+			}
+
+			message := InterfaceToString(requestJson["message"])
+			roomId := InterfaceToString(requestJson["room_id"])
+			senderId := InterfaceToString(requestJson["sender_id"])
+
+			row, _ := DbClient.GetChatroom(ctx, roomId)
+			if row.ID != roomId {
+				Response := make(map[string]interface{})
+				Response["request_type"] = "connect_to_chat_room"
+				Response["message"] = "chat room id is invalid"
+				responseBytes, _ := json.Marshal(Response)
+				WriteOnSocket(responseBytes, messageType, Conn)
+				break
+			}
+
+			hasUserId := false
+			for _, userId := range row.Users {
+				if userId == senderId {
+					hasUserId = true
+				}
+			}
+
+			if !hasUserId {
+				Response := make(map[string]interface{})
+				Response["request_type"] = "connect_to_chat_room"
+				Response["message"] = "sender_id is invalid"
+				responseBytes, _ := json.Marshal(Response)
+				WriteOnSocket(responseBytes, messageType, Conn)
+				break
+			}
+
+			if _, ok := ChatRoomList[roomId]; !ok {
+				Response := make(map[string]interface{})
+				Response["request_type"] = "send_message"
+				Response["message"] = "invalid room id"
+				responseBytes, _ := json.Marshal(Response)
+				WriteOnSocket(responseBytes, messageType, Conn)
+				return
+			}
+
+			thisChatRoom := ChatRoomList[roomId]
+			for _, client := range thisChatRoom.clients {
+				if client.connection != Conn {
+					Response := make(map[string]interface{})
+					Response["request_type"] = "send_message"
+					Response["message_from"] = senderId
+					Response["message"] = message
+					responseBytes, _ := json.Marshal(Response)
+					WriteOnSocket(responseBytes, messageType, client.connection)
+				}
+			}
+
+		default:
+			//	TODO: Handle unknown request and update client
+			log.Println("Unknown Method: ", requestJson["requestType"])
+			break
+		}
+	}
+
+}
+
+func WriteOnSocket(responseBytes []byte, messageType int, Conn *websocket.Conn) {
+
+	err := Conn.WriteMessage(messageType, responseBytes)
+	if err != nil {
+		log.Println("write:", err)
+	}
+}
+
+// InterfaceToString (interface{}) get value of an interface return as string {
+func InterfaceToString(ThisInterface interface{}) string {
+	if ThisInterface == nil {
+		fmt.Println("InterfaceToString :: NIL value passed for conversion")
+		return ""
+	}
+	switch ThisInterface.(type) {
+	case int:
+		return strconv.Itoa(ThisInterface.(int))
+	case int64:
+		return strconv.FormatInt(ThisInterface.(int64), 10)
+	case float64:
+		TmpStr := strconv.FormatFloat(ThisInterface.(float64), 'f', 10, 64)
+		if TmpStr[(len(TmpStr)-10):] == "0000000000" {
+			return TmpStr[:(len(TmpStr) - 11)]
+		}
+		return TmpStr
+	default:
+		return ThisInterface.(string)
+	}
+}
```

Comment:
this function converts interface to multiple types, why only use string in the function name?

Reply:
This function was unnecessary. Have deleted it.